src/System.Runtime.Extensions/src/System/AppDomain.cs

```diff
@@ -26,5 +27,187 @@ public partial class AppDomain
+        }
+
+        public int ExecuteAssembly(string assemblyFile) => ExecuteAssembly(assemblyFile, null, null, System.Configuration.Assemblies.AssemblyHashAlgorithm.None);
+        public int ExecuteAssembly(string assemblyFile, string[] args) => ExecuteAssembly(assemblyFile, args, null, System.Configuration.Assemblies.AssemblyHashAlgorithm.None);
+        public int ExecuteAssembly(string assemblyFile, string[] args, byte[] hashValue, System.Configuration.Assemblies.AssemblyHashAlgorithm hashAlgorithm)
+        {
+            if(assemblyFile == null)
+                throw new ArgumentNullException("assemblyFile");
+            /* TODO Assembly.LoadFrom is not implemented yet
+            Assembly assembly = Assembly.LoadFrom(path, hashValue, hashAlgorithm);
+            return ExecuteAssembly(assembly, args);
+            */
+            return 0;
+        }
+        private int ExecuteAssembly(Assembly assembly, string[] args)
+        {
+            MethodInfo entry = assembly.EntryPoint;
+            if(entry == null)
+                throw new MissingMethodException();
+            Object result = null;
+            int intResult = 0;
+            try {
+                if (entry.GetParameters().Length > 0)
+                    result = entry.Invoke(null, new object[] { args });
+                else
+                    result = entry.Invoke(null, null);
+            }
+            catch (TargetInvocationException targetInvocationException)
+            {
+                if (targetInvocationException.InnerException == null)
+                    throw;
+                //We are catching the TIE here and throws the inner exception only,
+                //this is needed to have a consistent exception story with desktop clr
+                ExceptionDispatchInfo.Capture(targetInvocationException.InnerException).Throw();
+            }
+            if(result != null)
+                intResult = (int) result;
+            return intResult;
+        }
+
+        public int ExecuteAssemblyByName(System.Reflection.AssemblyName assemblyName, params string[] args) 
+        {
+            Assembly assembly = Assembly.Load(assemblyName);
+            return ExecuteAssembly(assembly, args);            
+        }
+
+        public int ExecuteAssemblyByName(string assemblyName) => ExecuteAssemblyByName(assemblyName, null);
+        public int ExecuteAssemblyByName(string assemblyName, params string[] args)
+        {
+            Assembly assembly = Assembly.Load(assemblyName);
+            return ExecuteAssembly(assembly, args);
+        }
+
+        public object GetData(string name) => AppContext.GetData(name);
+        public void SetData(string name, object data) => AppContext.SetData(name, data);
+
+        public bool? IsCompatibilitySwitchSet(string value) 
+        {
+            bool result;
+            if(AppContext.TryGetSwitch(value, out result))
+                return result;
+            else
+                return new bool?();
```

Comment:
Nit: the `new bool?()` looks a little strange to me... It might be nicer as:

```c#
bool result;
return AppContext.TryGetSwitch(value, out result) ? result : default(bool?);
```